pkg/scheduler/core/generic_scheduler.go

```diff
@@ -253,23 +253,37 @@ func (g *genericScheduler) numFeasibleNodesToFind(numAllNodes int32) (numNodes i
 // Filters the nodes to find the ones that fit the pod based on the framework
 // filter plugins and filter extenders.
 func (g *genericScheduler) findNodesThatFitPod(ctx context.Context, prof *profile.Profile, state *framework.CycleState, pod *v1.Pod) ([]*v1.Node, framework.NodeToStatusMap, error) {
+	filteredNodesStatuses := make(framework.NodeToStatusMap)
+
 	// Run "prefilter" plugins.
 	s := prof.RunPreFilterPlugins(ctx, state, pod)
 	if !s.IsSuccess() {
-		return nil, nil, s.AsError()
+		if !s.IsUnschedulable() {
```

Comment:
I rethink it a bit.

The primary benefit of returning a FitError (instead of Error) is to continue preemption. Given that current preemption logic doesn't proceed upon `UnschedulableAndUnresolvable` status:

https://github.com/kubernetes/kubernetes/blob/21953d15ea48972f20a8de29d58bd5ce6d913914/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go#L224-L228

(but we can argue that custom preemption plugin may proceed even with `UnschedulableAndUnresolvable` status)

So maybe we can optimize the logic here:

```go
if s.Code() == UnschedulableAndUnresolvable {
```

Reply:
Not only to execute preemption but also proper metric reporting, so I think we should keep it this way and optimize later the fact that all nodes will have the same status.

Reply:
The question here is whether PreFilter plugins could return `Unschedulable`, as opposed to only `UnschedulableAndUnresolvable`.

I don't think we need to limit them.

Reply:
> so I think we should keep it this way and optimize later the fact that all nodes will have the same status.

That for sure has an improvement room, like simply using a `*` to represent all nodes.

> I don't think we need to limit them.

+1. I believe it's possible that some PreFilter plugin may return `Unschedulable` as well.